edgelet/iotedge/src/check/checks/identity_certificate_expiry.rs

```diff
@@ -0,0 +1,150 @@
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+
+use failure::{self, Context, ResultExt};
+
+use edgelet_core::{self, AttestationMethod, ManualAuthMethod, Provisioning, RuntimeSettings};
+
+use crate::check::{checker::Checker, Check, CheckResult};
+
+#[derive(Default, serde_derive::Serialize)]
+pub struct IdentityCertificateExpiry {
+    // provisioning: Option<Provisioning>, /* Note: contains shared keys */
```

Comment:
We do want to know the kind of provisioning, so add a `provisioning_mode: Option<&'static str>` and set it to `Some("manual")` or `Some("dps")` or `Some("external")`.